realsense2_camera/test/live_camera/rosci.py

```diff
@@ -0,0 +1,147 @@
+
+hub_reset = False
+logdir = None
+handle = None
+def usage():
+    ourname = os.path.basename( sys.argv[0] )
+    print( 'Syntax: ' + ourname + ' [options] [dir]' )
+    print( '        dir: location of executable tests to run' )
+    print( 'Options:' )
+    print( '        --debug              Turn on debugging information (does not include LibRS debug logs; see --rslog)' )
+    print( '        -v, --verbose        Errors will dump the log to stdout' )
+    print( '        -q, --quiet          Suppress output; rely on exit status (0=no failures)' )
+    print( '        -s, --stdout         Do not redirect stdout to logs' )
+    print( '        -r, --regex          Run all tests whose name matches the following regular expression' )
+    print( '        --list-tests         Print out all available tests. This option will not run any tests' )
+    print( '        --repeat <#>         Repeat each test <#> times' )
+    print( '        --config <>          Ignore test configurations; use the one provided' )
+    print( '        --device <>          Run only on the specified devices; ignore any test that does not match (implies --live)' )
+    print( '        --no-reset           Do not try to reset any devices, with or without a hub' )
+    print( '        --hub-reset          If a hub is available, reset the hub itself' )
+    print( '        --skip-disconnected  Skip live test if required device is disconnected (only applies w/o a hub)' )
+    print( 'Examples:' )
+    print( 'Running: python3.10 rosci.py -s' )
+    print( '    Runs all tests, but direct their output to the console rather than log files' )
+    print( 'Running: python3.10 rosci.py --list-tests' )
+    print( "    Will find all tests and print" )
+
+    sys.exit( 2 )
+
+def command(dev_name):
+       cmd =  ['pytest-3']
+       cmd += ['-s']
+       cmd += ['-m', ''.join(dev_name)]
+       cmd += ['-k', 'test_camera_imu_tests'] 
+       #cmd += ['-m', 'd415']
+       cmd += [''.join(dir_live_tests)]
+       cmd += ['--debug']
+       return cmd
+
+def run_test(cmd, dev_name, stdout=None, append =False):
+        handle = None
+        try:
+            stdout = stdout + os.sep + str(dev_name) + '_' + "test_camera_imu_tests" + ".log"
+            if stdout is None:
+                sys.stdout.flush()
+            elif stdout and stdout != subprocess.PIPE:
+                if append:
+                    handle = open( stdout, "a" )
+                    handle.write(
+                        "\n----------TEST-SEPARATOR----------\n\n" )
+                    handle.flush()
+                else:
+                    handle = open( stdout, "w" )
+
+            result = subprocess.run( cmd,
+                    stdout=handle,
+                    stderr=subprocess.STDOUT,
+                    universal_newlines=True,
+                    timeout=200,
+                    check=True )
+            if not result.returncode:
```

Comment:
What if we get an error code?

Reply:
Any error code !=0 invokes exception and thereby CI fails